src/common/containers/build-details.js

```diff
@@ -5,18 +5,46 @@ import Helmet from 'react-helmet';
 import BuildRow from '../components/build-row';
 import BuildLog from '../components/build-log';
 import { Message } from '../components/forms';
-import { fetchSnap } from '../actions/snap-builds';
+
+import { setGitHubRepository } from '../actions/repository-input';
+import { fetchBuilds, fetchSnap } from '../actions/snap-builds';
 
 import styles from './container.css';
 
 class BuildDetails extends Component {
 
+  fetchData({ snapLink, repository }) {
+    if (snapLink) {
+      this.props.dispatch(fetchBuilds(snapLink));
+    } else if (repository) {
+      this.props.dispatch(fetchSnap(repository.url));
+    }
+  }
+
   componentWillMount() {
-    this.props.dispatch(fetchSnap(this.props.fullName));
+    if (!this.props.repository && this.props.fullName) {
+      this.props.dispatch(setGitHubRepository(this.props.fullName));
```

Comment:
To properly propagate `repository` reducer this bit is needed (and duplicated in repository setup, builds list and build details). 

I wonder if it would be worth having additional layer of container/decorator that would trigger the action (to propagate data in reducer) and pass it to component that needs it.

But it sounds a little bit over-engineered...